### PR TITLE
Bump github.com/golangci/golangci-lint from 1.55.2 to 1.59.1 and fix linter errors

### PR DESCRIPTION
**Pull request by FlamingSaint**
## Which problem is this PR solving?
- Upgrade in #5564 is failing on linter errors

## Description of the changes
- fix lint errors

## How was this change tested?
- `make lint` `make test`

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: .github/actions/setup-node.js/action.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ runs:
       run: |
         echo "JAEGER_UI_NODE_JS_VERSION=$(cat jaeger-ui/.nvmrc)" >> ${GITHUB_ENV}
 
-    - uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v4.0.2
+    - uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c
```
Comment (yurishkuro) [inline .github/actions/setup-node.js/action.yml:11]: you have a bad merge

File: renovate.json [hunk 2]
```diff
@@ -7,4 +7,4 @@
     "changelog:dependencies"
   ],
   "suppressNotifications": ["prEditedNotification"]
-}
+}
```
Comment (yurishkuro) [inline renovate.json:10]: please tune your IDE/editor to not remove final new-line

Comment (FlamingSaint) [reply renovate.json:10]: Okay.

File: cmd/internal/flags/admin_test.go [hunk 3]
```diff
@@ -56,7 +56,7 @@ func TestAdminServerHandlesPortZero(t *testing.T) {
 	onlyEntry := message.All()[0]
 	hostPort := onlyEntry.ContextMap()["http.host-port"].(string)
 	port, _ := strconv.Atoi(strings.Split(hostPort, ":")[3])
-	assert.Greater(t, port, 0)
+	assert.Positive(t, port, 0)
```
Comment (yurishkuro) [inline cmd/internal/flags/admin_test.go:59]: 0 arg is not needed?

File: .github/actions/setup-node.js/action.yml (hunk 1 above)
Comment (FlamingSaint) [reply .github/actions/setup-node.js/action.yml]: I have fixed the merge.

File: pkg/tenancy/grpc.go [hunk 4]
```diff
@@ -100,7 +100,7 @@ func tenantFromMetadata(md metadata.MD, tenancyHeader string) (string, error) {
 // NewGuardingUnaryInterceptor blocks handling of RPCs whose tenancy header doesn't meet tenancy requirements.
 // It also ensures the tenant is directly in the context, rather than context metadata.
 func NewGuardingUnaryInterceptor(tc *Manager) grpc.UnaryServerInterceptor {
-	return func(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (any, error) {
+	return func(ctx context.Context, req any, _ /* info */ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (any, error) {
```
Comment (yurishkuro) [inline pkg/tenancy/grpc.go:103]: `/* info */` unnecessary

File: plugin/storage/cassandra/dependencystore/storage_test.go [hunk 5]
```diff
@@ -268,5 +268,5 @@ func TestGetBuckets(t *testing.T) {
 }
 
 func matchEverything() any {
-	return mock.MatchedBy(func(v []any) bool { return true })
+	return mock.MatchedBy(func(_ /* v */ []any) bool { return true })
```
Comment (yurishkuro) [inline plugin/storage/cassandra/dependencystore/storage_test.go:271]: `v` unnecessary

Comment (yurishkuro) [reply plugin/storage/cassandra/dependencystore/storage_test.go:271]: and `_`

File: plugin/storage/es/spanstore/writer.go [hunk 6]
```diff
@@ -117,7 +117,7 @@ func getSpanAndServiceIndexFn(archive, useReadWriteAliases bool, prefix, spanDat
 	spanIndexPrefix := prefix + spanIndex
 	serviceIndexPrefix := prefix + serviceIndex
 	if archive {
-		return func(date time.Time) (string, string) {
+		return func(_ /* date */ time.Time) (string, string) {
```
Comment (yurishkuro) [inline plugin/storage/es/spanstore/writer.go:120]:
```suggestion
		return func(_ time.Time) (string, string) {
```

File: plugin/storage/cassandra/dependencystore/storage_test.go (hunk 5 above)
Comment (FlamingSaint) [reply plugin/storage/cassandra/dependencystore/storage_test.go:271]: I thought the `_` would be better for readability based on last times PR ?

File: storage/spanstore/downsampling_writer_test.go [hunk 7]
```diff
@@ -81,7 +80,6 @@ func TestDownSamplingWriter_hashBytes(t *testing.T) {
 	}
 	_, _ = span.TraceID.MarshalTo(h.buffer)
 	// Same traceID should always be hashed to same uint64 in DownSamplingWriter.
-	assert.Equal(t, h.hashBytes(), h.hashBytes())
```
Comment (yurishkuro) [inline storage/spanstore/downsampling_writer_test.go:84]: The function comment is sensible, but the original test makes no sense, it didn't do that. Maybe fix it?

File: plugin/storage/cassandra/dependencystore/storage_test.go (hunk 5 above)
Comment (yurishkuro) [reply plugin/storage/cassandra/dependencystore/storage_test.go:271]: `_` is ok-ish

**Comment by yurishkuro** [comment]
I modified the hasher test a bit. Also, we're getting warnings from the new linter:
```
WARN [config_reader] The configuration option `run.skip-files` is deprecated, please use `issues.exclude-files`.
WARN [config_reader] The configuration option `run.skip-dirs` is deprecated, please use `issues.exclude-dirs`.
WARN [config_reader] The configuration option `run.skip-dirs-use-default` is deprecated, please use `issues.exclude-dirs-use-default`.
WARN [config_reader] The configuration option `linters.gosimple.go` is deprecated, please use global `run.go`.
```

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**